File: JavaInterpreter/src/CPLJavaInterpreter/Token.java
package CPLJavaInterpreter;

public class Token {
    private int rowNumber;
    private int columnNumber;
    private String lexeme;
    private TokenType tokType;

    public Token(int rowNumber, int columnNumber, String lexeme, TokenType tokType){
        if (rowNumber <= 0)
            throw new IllegalArgumentException("Invalid row number argument");
        if (columnNumber <= 0)
            throw new IllegalArgumentException("Invalid column number argument");
        if (lexeme == null || lexeme.length() == 0)
            throw new IllegalArgumentException("Invalid lexeme argument");
        if (tokType == null)
            throw new IllegalArgumentException("Invalid TokenType argument");

        this.rowNumber = rowNumber;
        this.columnNumber = columnNumber;
        this.lexeme = lexeme;
        this.tokType = tokType;
    }

    public int getRowNumber(){
        return rowNumber;
    }

    public int getColumnNumber(){
        return columnNumber;
    }

    public String getLexeme(){
        return lexeme;
    }

    public TokenType getTokType(){
        return tokType;
    }
}

File: Python Interpreter/PythonInterpreter/Token.py
import Enumerated


class Token:

    rowNumber = 0
    colNumber = 0
    tokType = Enumerated.TokenType
    lexeme = ""

    def __init__(self, rowNumber, colNumber, lexeme, tokType):
        if rowNumber <= 0:
            raise AttributeError("Invalid row number argument")
        if colNumber <= 0:
            raise AttributeError("Invalid column number argument")
        if lexeme is None or len(lexeme) == 0:
            raise AttributeError("Invalid lexeme argument")
        if tokType is None:
            raise AttributeError("Invalid TokenType argument")

        self.rowNumber = rowNumber
        self.colNumber = colNumber
        self.lexeme = lexeme
        self.tokType = tokType

    def get_row_number(self):
        return self.rowNumber

    def get_col_number(self):
        return self.colNumber

    def get_lexeme(self):
        return self.lexeme

    def get_tok_type(self):
        return self.tokType

